corehq/apps/auditcare/utils/export.py

```diff
@@ -70,6 +68,48 @@ def get_all_log_events(start_date=None, end_date=None):
     )
 
 
+def get_domain_first_access_times(domains, start_date=None, end_date=None):
+    """Query NavigationEventAudit events for _first event matching any of
+    `domains` within each authenticated session_.
+
+    NOTE: This function does _not_ query couch.
+
+    NOTE: This function may return multiple "access events" from the same
+          session (if multiple `domains` were accessed in the same session).
+
+    Resulting SQL query:
+
+    ```sql
+    SELECT
+        "user",
+        domain,
+        MIN(event_date) AS access_time
+    FROM auditcare_navigationeventaudit
+    WHERE (
+        domain IN ( {domains} )
+        AND event_date > {start_date}
+        AND event_date <= {end_date}
+        AND "user" IS NOT NULL
+        AND session_key IS NOT NULL
+    )
+    GROUP BY ("user", domain, session_key)
+    ORDER BY access_time ASC;
+    ```
+    """
+    sql_start_date = determine_sql_start_date(start_date)
+    where = get_date_range_where(sql_start_date, end_date)
+    where["domain__in"] = domains
+    where["user__isnull"] = False
+    where["session_key__isnull"] = False
+    return (NavigationEventAudit.objects
+            .values("user", "domain", "session_key")  # GROUP BY fields
+            .annotate(access_time=Min("event_date"))
+            .values("user", "domain", "access_time")  # SELECT fields
```

Comment:
wow this must have taken some time to figure out. It's not at all obvious to me that the two different `.values` here would do different things like this. Maybe that's standard and it's just been a really long time since I've really used the django ORM for anything other than `.filter`. Thanks for these comments

Reply:
Yes, you can chain multiple `.values()` (I think there are other reasons you might do this besides specifying an alternative `SELECT` statement on a query with aggregations), but it's the only way to accomplish that here (at least on our version of Django).  Django v3.2 has a new `.alias()` method that might also apply here, but I didn't dig into those semantics when I realized I couldn't use it.